header.go

```diff
@@ -141,7 +141,7 @@ func ReadPackageHeader(r io.Reader) (*Header, error) {
 
 	// read the "store"
 	store := make([]byte, h.Length)
-	n, err = r.Read(store)
+	n, err = io.ReadFull(r, store)
 	if err != nil {
 		return nil, err
```

Comment:
If insufficient bytes are read into `store`, `io.ReadFull` will return `io.ErrUnexpectedEOF` and we will pass it back to the caller instead of the desired `ErrBadStoreLength`.

Looking at the implementation of [io.ReadFull](https://golang.org/src/io/io.go?s=10607:10673#L291), it seems that the only advantage it gives is a check that the required number of bytes was read. I already do this on #L150, so I'm not sure how your PR helps.

Can you please provide some explanation? Perhaps `r.Read(store)` is not reading enough bytes and we need the loop?